src/clj_kondo/impl/analyzer/test.clj

```diff
@@ -23,6 +23,13 @@
            body))))
    defined-by))
 
+(defn analyze-testing [ctx expr resolved-as-namespace]
+  (let [[testing testing-str & rest-children] (:children expr)
+        kns (keyword resolved-as-namespace)]
+    (common/analyze-children (assoc-in ctx [:context kns :testing-str] (utils/sexpr testing-str))
```

Comment:
I think you need to use the ctx with the context while analyzing the testing call.

Reply:
@borkdude you mean this?
```suggestion
    (common/analyze-children (assoc-in ctx [:context kns :testing-str] (utils/sexpr testing))
```
I tried that as well but it doesn't work

Reply:
No, that's not what I meant. You have to pass the ctx with the while analyzing the whole testing expression, not on the child.

Reply:
hum, so I should analyze the whole `expr` with that ctx, right?
I'll give a try

Reply:
Hum, I changed it but I still get the same empty context, I don't know what I'm missing here

Reply:
You have to include `:context true` or `:context [:keyword]` to receive the context in the analysis.

Reply:
OK, you already did that.

Reply:
I'll take a look later this week.

Reply:
@ericdallo Figured it out already. You didn't lint the children of testing.

```
(defn analyze-testing [ctx expr resolved-as-namespace]
  (let [[_ testing-str :as children] (:children expr)
        kns (keyword resolved-as-namespace)]
    (common/analyze-children (assoc-in ctx [:context kns :testing-str] (utils/sexpr testing-str))
                             children)))
```

Reply:
even with that, the context comes empty to me 🤔

Reply:
```
$ clojure -M:clj-kondo/dev --lint - --config '{:output {:analysis {:context true} :format :edn}}' <<< '(require (quote (clojure.test))) (clojure.test/testing "foo" (inc 1))' | jet --pretty
```

```
                         {:fixed-arities #{1},
                          :end-row 1,
                          :name-end-col 66,
                          :name-end-row 1,
                          :name-row 1,
                          :name inc,
                          :filename "<stdin>",
                          :from user,
                          :col 62,
                          :name-col 63,
                          :context {:clojure.test {:testing-str "foo"}},
                          :end-col 69,
                          :arity 1,
                          :row 1,
                          :to clojure.core}
```

Reply:
Yeah, you are right, so probably my test is wrong somewhere 🤔

Reply:
Oh, there is something weird happening indeed:
```clojure
{{:end-row 1,
                          :name-end-col 55,
                          :name-end-row 1,
                          :name-row 1,
                          :name testing,
                          :filename "<stdin>",
                          :from user,
                          :macro true,
                          :col 35,
                          :name-col 35,
                          :end-col 55,
                          :varargs-min-arity 1,
                          :row 1,
                          :to clojure.test}
                         {:fixed-arities #{1},
                          :end-row 1,
                          :name-end-col 66,
                          :name-end-row 1,
                          :name-row 1,
                          :name inc,
                          :filename "<stdin>",
                          :from user,
                          :col 62,
                          :name-col 63,
                          :context {:clojure.test {:testing-str "foo"}},
                          :end-col 69,
                          :arity 1,
                          :row 1,
                          :to clojure.core}
                         {:end-row 1,
                          :name-end-col 55,
                          :name-end-row 1,
                          :name-row 1,
                          :name testing,
                          :filename "<stdin>",
                          :from user,
                          :macro true,
                          :col 34,
                          :name-col 35,
                          :end-col 70,
                          :arity 2,
                          :varargs-min-arity 1,
                          :row 1,
                          :to clojure.test}]}}

```

Reply:
The issue is that only children of testing get this context, but not the testing call itself. I need to re-organize some stuff in `analyze-call` to support that. It's already supported for custom hooks.

Reply:
Also: I think you might want to store the stack of testing-strs, not only one maybe?

Reply:
I see.
I thought about leaving that on LSP side, since we will already need to build the tree there, not sure adding there would help that much

Reply:
I'll probably know more about it when start to implement on clojure-lsp

Reply:
That's ok, adding less data is better for performance, we will see.